src/main/java/com/google/maps/TextSearchRequest.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2015 Google Inc. All rights reserved.
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+ * ANY KIND, either express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.google.maps;
+
+import com.google.gson.FieldNamingPolicy;
+import com.google.maps.errors.ApiException;
+import com.google.maps.internal.ApiConfig;
+import com.google.maps.internal.ApiResponse;
+import com.google.maps.model.LatLng;
+import com.google.maps.model.PlaceDetails;
+import com.google.maps.model.PlacesSearchResponse;
+import com.google.maps.model.PlacesSearchResult;
+
+/**
+ * A <a href="https://developers.google.com/places/web-service/search#TextSearchRequests">Text Search</a> request.
+ */
+public class TextSearchRequest
+    extends PendingResultBase<PlacesSearchResponse, TextSearchRequest, TextSearchRequest.Response>{
+
+  static final ApiConfig API_CONFIG = new ApiConfig("/maps/api/place/textsearch/json")
+      .fieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES);
+
+  protected TextSearchRequest(GeoApiContext context) {
+    super(context, API_CONFIG, Response.class);
+  }
+
+  /**
+   * query is the text string on which to search, for example: "restaurant".
+   */
+  public TextSearchRequest query(String query) {
+    return param("query", query);
+  }
+
+  /**
+   * location is the latitude/longitude around which to retrieve place information.
+   */
+  public TextSearchRequest location(LatLng location) {
+    return param("location", location);
+  }
+
+  /**
+   * radius defines the distance (in meters) within which to bias place results.
+   */
+  public TextSearchRequest radius(int radius) {
+    if (radius > 50000) {
+      throw new IllegalArgumentException("The maximum allowed radius is 50,000 meters.");
+    }
+    return param("radius", String.valueOf(radius));
+  }
+
+  /**
+   * minprice restricts to places that are at least this price level.
+   */
+  public TextSearchRequest minprice(PlaceDetails.PriceLevel priceLevel) {
```

Comment:
There's inconsistency here in the capitalisation - minprice/maxprice where later you've used minWidth/maxWidth.